client.go

```diff
@@ -132,9 +132,16 @@ func toClientOptions(c *Client, o *clientOptions) *mqtt.ClientOptions {
 		opts.SetClientID(o.clientID)
 	}
 
+	if o.tlsConfig != nil {
+		o.brokerAddress = fmt.Sprintf("tls://%s", o.brokerAddress)
+	} else {
```

Comment:
`else` pattern can be avoided, let's move this to its own function to format the address.

Reply:
Fixed 🙇